src/gpu/sycl/ref_shuffle.hpp

```diff
@@ -0,0 +1,117 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_SYCL_GENERIC_SHUFFLE_HPP
+#define GPU_SYCL_GENERIC_SHUFFLE_HPP
+
+#include "common/c_types_map.hpp"
+#include "common/utils.hpp"
+#include "gpu/gpu_shuffle_pd.hpp"
+#include "gpu/sycl/sycl_gpu_primitive.hpp"
+#include "gpu/sycl/sycl_io_helper.hpp"
+#include "gpu/sycl/sycl_post_ops.hpp"
+#include "gpu/sycl/sycl_primitive_conf.hpp"
+#include "gpu/sycl/sycl_q10n.hpp"
+#include "gpu/sycl/sycl_types.hpp"
+#include "sycl/sycl_stream.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace sycl {
+
+using namespace std;
+using namespace format_tag;
+using namespace utils;
+struct ref_shuffle_t : public sycl_gpu_primitive_t {
+    using sycl_gpu_primitive_t::sycl_gpu_primitive_t;
+
+    struct pd_t : public gpu_shuffle_pd_t {
+        using gpu_shuffle_pd_t::gpu_shuffle_pd_t;
+
+        DECLARE_COMMON_PD_T("dpcpp:ref:any", ref_shuffle_t);
+
+        status_t init(engine_t *engine) {
+            using namespace data_type;
+            const memory_desc_wrapper data_d(src_md(0));
+            const memory_desc_wrapper dst_d(dst_md(0));
+
+            const data_type_t data_type = src_md()->data_type;
+            bool ok = is_fwd()
```

Comment:
What is the reason for implementing only forward?

Reply:
Checked and updated